src/dom.js

```diff
@@ -55,7 +56,14 @@ export function waitForChild(parent, checkFunc, callback) {
  * @param {function()} callback
  */
 export function waitForBody(doc, callback) {
-  waitForChild(doc.documentElement, () => !!doc.body, callback);
+  // Prefer DocumentState.onBodyAvailable() to minimize creation of
+  // MutationObserver by waitForChild().
+  if (doc.defaultView) {
```

Comment:
I thought so too but I wasn't sure. [MDN](https://developer.mozilla.org/en-US/docs/Web/API/Document/defaultView) mentions that it may be null, but this may never happen in AMP. I'll remove it.